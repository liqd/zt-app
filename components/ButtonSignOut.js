import React from 'react'
import { View } from 'react-native'
import { Button } from '@rneui/base'
import { styles } from './ButtonSignOut.styles'
import {useAuthorization} from '../containers/Auth/AuthProvider.js'

export const ButtonSignOut = () => {
  const {signOut} = useAuthorization()
  return (
    <View style={styles.logoutContainer}>
      <Button
        onPress={signOut}
        title='Sign out'
        type='clear'
      />
    </View>
  )
}
