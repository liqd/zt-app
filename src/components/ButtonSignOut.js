import React from 'react'
import { useTranslation } from 'react-i18next'
import { Button } from '@rneui/base'

import {useAuthorization} from '../containers/Auth/AuthProvider.js'

import { styles } from './ButtonSignOut.styles'

export const ButtonSignOut = () => {
  const { t } = useTranslation()
  const {signOut} = useAuthorization()
  return (
    <Button
      buttonStyle={styles.logoutButton}
      onPress={signOut}
      title={t('Sign out')}
    />
  )
}
