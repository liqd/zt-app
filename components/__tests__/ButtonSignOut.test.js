import React from 'react'
import { ButtonSignOut } from '../ButtonSignOut'
import { render } from '@testing-library/react-native'

test('Test ButtonSignOut', () => {
  const { toJSON } = render(<ButtonSignOut />)
  expect(toJSON()).toMatchSnapshot()
})
