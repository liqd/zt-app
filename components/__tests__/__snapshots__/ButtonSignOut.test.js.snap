// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonSignOut 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "flex-end",
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 2,
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 2,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2089dc",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Sign out
        </Text>
      </View>
    </View>
  </View>
</View>
`;
